api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -72,41 +70,22 @@ func (mw *azureManagedControlPlaneWebhook) Default(ctx context.Context, obj runt
 	if !ok {
 		return apierrors.NewBadRequest("expected an AzureManagedControlPlane")
 	}
-	if m.Spec.NetworkPlugin == nil {
-		networkPlugin := "azure"
-		m.Spec.NetworkPlugin = &networkPlugin
-	}
-	if m.Spec.LoadBalancerSKU == nil {
-		loadBalancerSKU := "Standard"
-		m.Spec.LoadBalancerSKU = &loadBalancerSKU
-	}
-
-	if m.Spec.Version != "" && !strings.HasPrefix(m.Spec.Version, "v") {
-		normalizedVersion := "v" + m.Spec.Version
-		m.Spec.Version = normalizedVersion
-	}
 
-	if m.Spec.Identity == nil {
-		m.Spec.Identity = &Identity{
-			Type: ManagedControlPlaneIdentityTypeSystemAssigned,
-		}
-	}
+	setDefault[*string](&m.Spec.NetworkPlugin, ptr.To(AzureNetworkPluginName))
+	setDefault[*string](&m.Spec.LoadBalancerSKU, ptr.To("Standard"))
+	m.Spec.Version = setDefaultVersion(m.Spec.Version)
```

Comment:
Oof yeah it's pretty confusing to look at. It would be nice to get the generic `setDefault` to be compatible with non-pointers too. Some defaults also require more checks, so I think it's down to changing all of the type-specific defaults to passing in the field as a parameter.